src/components/engagement_filter_bar/components/advanced_search_input.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState, useRef, useEffect } from 'react';
+import {
+  ActionGroup,
+  Button,
+  Card,
+  CardBody,
+  Form,
+  FormGroup,
+  Popper,
+  SearchInput,
+  TextInput,
+} from '@patternfly/react-core';
+
+export interface AdvancedSearchInputProps {
+  searchValue:string,
+  categoryValue:string,
+  onSearchChange:(value:string) => void,
+  onSearch:(value:string, category:string) => void
+  onCategoryChange:(value:string) => void
+}
+export function AdvancedSearchInput({
+  searchValue,
+  categoryValue,
+  onSearchChange,
+  onSearch,
+  onCategoryChange,
+}: AdvancedSearchInputProps) {
+
+  const advancedFormRef = useRef(null);
+  const categoryRef = useRef(null);
+  const engagementRef = useRef(null);
+  const searchInputRef = useRef(null);
+
+  const [ isAdvancedSearchOpen, setIsAdvancedSearchOpen ] = useState<boolean>(false);
+
+  function handleSearch() {
+    setIsAdvancedSearchOpen(false);
+    onSearch(searchValue, categoryValue);
+  }
+
+  function handleEngagementBoxChange(searchTerm:string) {
+    const engagementSearch = categoryValue?.length === 0 ? searchTerm : `${searchTerm} category='${categoryValue}'`;
+    onSearchChange(engagementSearch);
+  }
+
+  const isInitialMount = useRef(true);
+  // After initial page load, whenever the advanced search menu is opened, the browser focus should be placed on the
+  // first advanced search form input. Whenever the advanced search menu is closed, the browser focus should
+  // be returned to the search input.
+  useEffect(() => {
+
+    if (isInitialMount.current) {
+      isInitialMount.current = false;
+    } else {
+      if (isAdvancedSearchOpen && categoryRef && categoryRef.current) {
+        categoryRef.current.focus();
+      } else if (!isAdvancedSearchOpen && searchInputRef) {
+        searchInputRef.current.focus();
+      }
+    }
+  }, [isAdvancedSearchOpen]);
+
+  useEffect(() => {
```

Comment:
We can probably accomplish this in a simpler way so we don't have to deal with adding and removing listeners manually.